compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2070,7 +2073,39 @@ impl<'a> Parser<'a> {
         if has_bang {
             self.expect(&token::Not)?; // `!`
         }
-        let ident = self.parse_ident()?;
+
+        let ident = match self.parse_ident() {
+            Ok(ident) => ident,
+            Err(mut err) => {
+                match (
+                    &self.token.kind,
+                    self.look_ahead(1, |token| token.ident()),
+                    self.look_ahead(2, |token| {
+                        token.kind == TokenKind::CloseDelim(Delimiter::Parenthesis)
+                    }),
+                ) {
+                    (
+                        TokenKind::OpenDelim(Delimiter::Parenthesis),
+                        Some((Ident { span, .. }, _)),
+                        true,
+                    ) => {
+                        err.span_note(
+                            span,
+                            "try removing the parenthesis around the name for this `macro_rules!`",
```

Comment:
I think it's better to suggest removing the parenthesis with `span_suggestion` or `span_suggestion_verbose`.

Reply:
Thank you, that does seem a lot better, I moved the span to include the parentheses now too.